profiles/schema.py

```diff
@@ -261,6 +273,31 @@ def resolve_phones(self, info, **kwargs):
     def resolve_addresses(self, info, **kwargs):
         return Address.objects.filter(profile=self)
 
+    def resolve_sensitivedata(self, info, **kwargs):
+        # FIXME:
+        # Since the serviceType argument gets eaten by the parent resolver in most cases you have to provide the
+        # serviceType also for this resolver in order to make permission control work correctly. Example:
+        #
+        # {
+        #   profile(id: "123", serviceType: BERTH) {
+        #     firstName
+        #     sensitivedata(serviceType: BERTH) {
+        #       ssn
+        #     }
+        #   }
+        # }
+        #
+        # It would be nice to get rid of this requirement sometime in the future.
+
+        service = Service.objects.filter(service_type=kwargs.get("serviceType")).first()
```

Comment:
have you investigated the possibility of passing this to the context in those more general resolvers, where you already have this argument. ie. something like this:
```
@staff_required(required_permission="view")
def resolve_profiles(self, info, **kwargs):
    service_type = kwargs.get("serviceType")
    info.context.service_type = service_type
    info.context.save() . # this might not be needed
    return Profile.objects.filter(
        service_connections__service__service_type=service_type
    )
```

then you would be able probably to avoid using the argument here?

Reply:
graphene's source code [hints that it should be possible](https://github.com/graphql-python/graphene/blob/8e7d76bbce3eb8755bf4f3beba8b9d8b026c6424/graphene/types/context.py#L14)

Reply:
I'll give that a try. At first that kind of approach seemed a bit hacky to me but maybe it's worth investigating more.